src/util/helpers/langDetect.js

```diff
@@ -0,0 +1,240 @@
+    {pattern: /\(.*\)( )*=>( )*.+/, points: 1},
+    // (else )if statement
+    {pattern: /(else )?if( )+\(.+\)/, points: 1},
+    // while loop
+    {pattern: /while( )+\(.+\)/, points: 1},
+    // C style variable declaration.
+    {pattern: /(^|\s)(char|long|int|float|double)( )+\w+( )*=?/, points: -1},
+    // pointer
+    {pattern: /(\w+)( )*\*( )*\w+/, points: -1},
+    // HTML <script> tag
+    {pattern: /<(\/)?script( type=('|")text\/javascript('|"))?>/, points: -50}
+  ],
+
+  'Python': [
+    // Function definition
+    {pattern: /def( )+\w+\(.*\)( )*:/, points: 2},
+    // while loop
+    {pattern: /while (.+):/, points: 2},
+    // from library import something
+    // eslint-disable-next-line
+    {pattern: /from [\w\.]+ import (\w+|\*)/, points: 2},
+    // class keyword
+    {pattern: /class( )*\w+(\(( )*\w+( )*\))?( )*:/, points: 2},
+    // if keyword
+    {pattern: /if( )+(.+)( )*:/, points: 2},
+    // elif keyword
+    {pattern: /elif( )+(.+)( )*:/, points: 2},
+    // else keyword
+    {pattern: /else:/, points: 2},
+    // for loop
+    {pattern: /for (\w+|\(?\w+,( )*\w+\)?) in (.+):/, points: 2},
+    // Python variable declaration.
+    {pattern: /\w+( )*=( )*\w+(?!;)(\n|$)/, points: 1},
+    // import something
+    // eslint-disable-next-line
+    {pattern: /import ([[^\.]\w])+/, points: 1, nearTop: true},
+    // print statement/function
+    {pattern: /print((( )*\(.+\))|( )+.+)/, points: 1},
+    // &&/|| operators
+    {pattern: /(&{2}|\|{2})/, points: -1}
+  ],
+
+  'HTML': [
+    {pattern: /<!DOCTYPE (html|HTML PUBLIC .+)>/, points: 2, nearTop: true},
+    // Tags
+    {pattern: /<[a-z0-9]+(( )*[\w]+=('|").+('|")( )*)?>.*<\/[a-z0-9]+>/g, points: 2},
+    // Properties
+    // eslint-disable-next-line
+    {pattern: /[a-z\-]+=("|').+("|')/g, points: 2},
+    // PHP tag
+    {pattern: /<\?php/, points: -50}
+  ],
+
+  'CSS': [
+    // Properties
+    // eslint-disable-next-line
+    {pattern: /[a-z\-]+:(?!:).+;/, points: 2},
+    // <style> tag from HTML
+    {pattern: /<(\/)?style>/, points: -50}
+  ],
+
+  'R': [
+    // undefined keyword for numbers
+    {pattern: /NaN/g, points: 5},
+    // undefined keyword for strings
+    {pattern: /NA/g, points: 5},
+    // print statement
+    {pattern: /print\(.*\)/, points: 1},
+    // Variable declaration
+    {pattern: /\w+ ?<-/, points: 10},
+    // Vector declaration
+    {pattern: /c\(.*\)/, points: 5},
+    // == operator
+    {pattern: /==/g, points: 1},
+    // != operator
+    {pattern: /!=/g, points: 1},
+    // Function definition
+    {pattern: /function ?\((.*\n)*?\n?[\t ]*{/g, points: 1},
+    // null keyword
+    {pattern: /NULL/g, points: 2},
+    // (else )if statement
+    {pattern: /((else )|([\n \t]+))if ?\(!?.+\)\n?[\t ]*{?/, points: 2},
+    // while loop
+    {pattern: /while\(.+\).*\n?[\t ]*{/, points: 1}
+  ],
+
+  'Unknown': []
+}
+
+function getPoints (language, lineOfCode, checkers) {
+  return _.reduce(_.map(checkers, function (checker) {
```

Comment:
We should write this in ES6 so we can drop the underscore library.
But lets get the tests working first and then we will see